### ISSUE TITLE
Support automatic .NET SDK version bumps within the ``github-actions`` update checker.

### ISSUE THREAD

**Issue by AraHaan** [dependabot/dependabot-core#3515]
Basically I wish for dependabot to automatically update the .NET SDK that is specifically told to be used with ``setup-dotnet`` so that way I do not have to manually update the file myself and try to keep track with .NET SDK releases (like for example needing to update from .NET SDK 6 Preview 1 / 2 -> Preview 3).

If Dependabot could version bump that probably could be scraped from https://dotnet.microsoft.com/download/dotnet, I think dependabot would then keep track of the .NET SDK updates that are then released publicly the same way it updates each individual github action in the workflow files.

So if this becomes a think it will help simplify the hard things associated with manually updating the file from the above.

However since some people prefer to always prefer stable, while others might prefer the preview .NET SDKs I think an option could be added to the ``github-actions`` setting in dependabot.yml to tell it that they prefer stable / preview releases to the .NET SDK (only if a preview release is available, else fall back to latest stable).

With that in mind it would reduce the maintenance cost of manually updating the workflows just to update the .NET SDK.

Besides, it's ok to always update the .NET SDK (even if they do not target .NET 6 yet but use the SDK to prevent the nuget restore bugs on the .NET 5 SDKs).

**Comment by xt0rted** [comment]
If you don't specify a version then the `setup-dotnet` action loads the sdk version from your `global.json` This is how my projects are setup. I'd like to see dependabot support bumping the `global.json` instead of the workflow file which gives the same end results and means your local dev environment will use the same version.

**Comment by AraHaan** [comment]
True, but then if you want it to automatically pickup the prerelease versions it would have to bump those automatically vs bumping the ``x.y`` when the ``prerelease`` flag on that action is set to true (to tell the update script to install the latest prerelease version).

**Comment by martincostello** [comment]
Just found this while checking for existing issues while opening a new issue. FYI, I maintain a GitHub Action that can update the SDK version in `global.json` for you: [martincostello/update-dotnet-sdk](https://github.com/martincostello/update-dotnet-sdk). It also supports pre-releases _and_ the .NET CI system itself out of the dotnet/install repo.

Sure, it's not as easy as dependabot doing it for you, but it's a way.

**Comment by JamieMagee** [comment]
@AraHaan I think this is complete as Dependabot now supports updating .NET SDK versions in `global.json`, and `actions/setup-dotnet` automatically installs the version defined in `global.json`. This includes pre-release versions.

See:
- https://github.blog/changelog/2024-11-19-dependabot-can-now-perform-version-updates-for-the-net-sdk/
- https://devblogs.microsoft.com/dotnet/using-dependabot-to-manage-dotnet-sdk-updates/
